packages/@aws-cdk/aws-lambda/lib/log-retention-provider/index.ts

```diff
@@ -2,15 +2,16 @@
 
 // eslint-disable-next-line import/no-extraneous-dependencies
 import * as AWS from 'aws-sdk';
+import { LogRetentionRetryOptions } from '../log-retention';
```

Comment:
I don't think this will work, the lambda function execution should fail. Did you try deploying a stack that used this?

This file gets deployed as a lambda function during stack deployment. The CDK code is not available there and it should not be.

Reply:
I linked it using a postinstall hook in my local test project. Then it works, but this is indeed incorrect.